app/views/PillarAnalysis/AnalyticalStatementInput/index.tsx

```diff
@@ -320,6 +320,54 @@ function AnalyticalStatementInput(props: AnalyticalStatementInputProps) {
         ],
     );
 
+    const handleAnalyticalEntryUp = useCallback(
+        (dropOverEntryClientId?: string) => (
+            onFieldChange(
+                (oldEntries: AnalyticalEntry) => {
+                    const { entries } = value;
+                    const fromIndex = entries?.findIndex(
+                        (entry) => entry.clientId === dropOverEntryClientId,
+                    );
+                    if (isNotDefined(fromIndex)) {
+                        return oldEntries;
+                    }
+
+                    const modifyingEntries = [...(oldEntries ?? [])];
+                    const element = modifyingEntries[fromIndex];
+                    const toIndex = fromIndex - 1;
+                    modifyingEntries.splice(fromIndex, 1);
+                    modifyingEntries.splice(toIndex, 0, element);
```

Comment:
double check this logicc

Reply:
After removing element from the array the remaining element does not shift automatically so its safe to insert element right after removing it.